### DISCUSSION TITLE
Problematic 3D Delaunay triangulation !

### DISCUSSION THREAD

**Discussion by vzavec** [pyvista/pyvista#5579] [Q&A] [answered]
Attached is a IGES file with half the hull of the ship. First both datasets were merged and cleaned.
Creation a mesh with delaunay_3d was not successful. 
I need a calculated clipped volume with the given plane in the code bellow.

Is Pyvista capable to generate a 3D tetrahedral mesh from a surface as specified in my IGES file?

Please for advice to overcome this problem!

Plotted hull after clipping:

<img width="423" alt="hull" src="https://github.com/pyvista/pyvista/assets/20023991/071ac6c5-3518-471a-adf9-cb87321d7582">

My half hull file:
[half_hull.zip](https://github.com/pyvista/pyvista/files/14167361/half_hull.zip)

Python  code:

```python
import pyvista as pv
import pyiges

# here we have only one half hull
iges = pyiges.read("half_hull.IGES")

# convert all surfaces to a vtk mesh
msh = iges.to_vtk(bsplines=False, surfaces=True, merge=True, delta=0.05)
# another half hull
msh2 = iges.to_vtk(bsplines=False, surfaces=True, merge=True, delta=0.05)
msh2.points[:,1] *= -1

# Merge both datasets
surf = pv.merge([msh, msh2]).clean()  

# scale and move to new position
surf = surf.scale([40.0, 40.0, 40.0], inplace=True)
surf = surf.translate((117.08, 0, 14.31), inplace=True)

# plane orientation and center
direction = [0, 0, 1]
center = (0, 0, 13.0)
plz = pv.Plane( center=center, direction=direction )

# clip hull with plane plz 
sclipp = surf.clip_surface(plz, invert=True)
sclippf = surf.clip_surface(plz, invert=False)


# bad result with
#surf = sclipp.delaunay_3d(alpha=10)

# bad result for volume
print("Hull volume (m3): ", sclipp.volume)
print("Mass center: ", sclipp.center_of_mass())

pl = pv.Plotter()
pl.add_mesh(sclippf, style='wireframe')   
pl.add_mesh(sclipp, color="brown", opacity=.35) 
pl.show_grid()
pl.show()
```

**Comment by tkoyama010** [comment, marked as answer]
Maybe [tetgen](https://github.com/pyvista/tetgen) is what you are looking for. The other solution between PyVista and Gmsh, [pvgmsh](https://github.com/pyvista/pvgmsh), is being developed, but this is still being experimented with.

**Reply by vzavec** [reply]
Sorry to hear that. Having spent a lot of time studying the PYVISTA.
Thanks anyway!
